core/node/genesis/src/lib.rs

```diff
@@ -185,53 +187,99 @@ pub fn mock_genesis_config() -> GenesisConfig {
         fee_account: Default::default(),
         dummy_verifier: false,
         l1_batch_commit_data_generator_mode: Default::default(),
+        custom_genesis_state_path: None,
     }
 }
 
-// Insert genesis batch into the database
-pub async fn insert_genesis_batch(
+pub fn make_genesis_batch_params(
+    storage_logs: &[StorageLog],
+    base_system_contract_hashes: BaseSystemContractsHashes,
+    protocol_version: ProtocolVersionId,
+) -> (GenesisBatchParams, L1BatchCommitment) {
+    // This action disregards how leaf indeces used to be ordered before, and it reorders them by
+    // sorting by <address, key>, which is required for calculating genesis parameters.
+    let storage_logs = get_deduped_log_queries(storage_logs)
```

Comment:
`get_deduped_log_queries` is called twice, here and in `insert_factory_deps_and_storage_logs`. This is a) suboptimal; b) error-prone. Please refactor it. One way to do it is to accept deduplicated_writes in both `make_genesis_batch_params` and `insert_factory_deps_and_storage_logs` and deduplicate them on caller side